### ISSUE TITLE
Implement Comment entity with controller, dto, service and repository.

### ISSUE THREAD

**Issue by oleksandr-jr** [JR-Abstract/reddit-clone-abstract#12]
## Description

Implement mapping from entity to DTO's and back using map struct library.

## Related to

Only after #3 

### Controller:

`@RequestMapping("/api/v1/comments")`

```java 
    @PostMapping
    public ResponseEntity<Void> createComment(@RequestBody CommentsDto commentsDto)

    @GetMapping("/by-post/{postId}")
    public ResponseEntity<List<CommentsDto>> getAllCommentsForPost(@PathVariable Long postId)

    @GetMapping("/by-user/{userName}")
    public ResponseEntity<List<CommentsDto>> getAllCommentsForUser(@PathVariable String userName)

```

### CommentsDto

```java
    private Long id;
    private Long postId;
    private Instant createdDate;
    private String text;
    private String userName;
```

### Service
``` java
    public void save(CommentsDto commentsDto)

    public List<CommentsDto> getAllCommentsForPost(Long postId)

    public List<CommentsDto> getAllCommentsForUser(String userName)
```

## Acceptance criteria
AC1: The controller, DTO, service and repository is implemented
AC2: The controller is mapped to `/api/v1/comments` 
AC3: Create post, `/by-post/{postId}` and `/by-user/{userName}` endpoints work
AC4: Test are implemented with code coverage > 70%

**Comment by oleksandrhavryush** [comment]
Create Comment entity class with fields, annotations, and relations

**Comment by oleksandrhavryush** [comment]
Create CommentDTO entity class with fields, annotations, and relations

**Comment by oleksandrhavryush** [comment]
Create CommentMapper entity class with fields, annotations and relations
Refactor name CommentDto into CommentDTO

**Comment by oleksandrhavryush** [comment]
Create CommentRepository interface with methods

**Comment by oleksandrhavryush** [comment]
Create CommentService entity class with methods and annotations
Add one method findByUserUsername to CommentRepository interface

**Comment by oleksandrhavryush** [comment]
Create CommentController entity class with methods and annotations
Refactor in CommentMapper and CommentRepository interfaces

**Comment by oleksandrhavryush** [comment]
Completed the task, but encountered an issue in two methods:

public ResponseEntity<List<CommentDTO>> getAllCommentsForPost(@PathVariable Long postId)
1. Failed to convert value of type 'java.lang.String' to required type 'java.lang.Long'; For input string: "postId"

public ResponseEntity<List<CommentDTO>> getAllCommentsForUser(@PathVariable String userName)
2. Failed to convert value of type 'java.lang.String' to required type 'java.lang.Long'; For input string: "userName"

Further investigation and resolution are needed for handling the conversion from String to Long in these cases.

**Comment by aleos-dev** [comment]
@oleksandrhavryush @oleksandr-jr 

1. The RequiredArgsConstructor annotation from Lombok is intended to generate a constructor with required arguments, which are fields marked as final or with the NonNull annotation.

![Image](https://github.com/JR-Abstract/reddit-clone-abstract/assets/121808229/2e61861a-53b0-4c00-8bc8-04c4924afdb6)

2. We have a problem that @InheritInverseConfiguration sets up the id, and we can simply delete our previous comment. Therefore, the mapping for id should work only in one direction in our case (or we should check it manually). However, if we want the option to edit a comment, we should figure out how to implement it properly.
    
![Image](https://github.com/JR-Abstract/reddit-clone-abstract/assets/121808229/01d34603-3d1d-4436-bb15-9c58ebd8828d)

3. When we request comments for a non-existent username or post, we currently receive an empty list. Should we not instead return a warning that such entities do not exist? This issue also occurs in other controllers, so we should standardize our approach.

Overall, if we ensure that the fields are injected, our intended functionality works fine.

**Comment by oleksandr-jr** [comment]
@aleos-dev @oleksandrhavryush  
1. Fully agree.
2. Yes, let's create a ticket for that.
3. It's ok to return empty list

**Comment by oleksandr-jr** [comment]
@oleksandrhavryush Please, create a pull request into dev and link it to this ticket.

**Comment by oleksandr-jr** [comment]
@oleksandrhavryush And one more thing, the changes are not merged into integration and I see some conflicts during merging.
Please check the it.
